examples/iiif.js

```diff
@@ -0,0 +1,46 @@
+import Map from '../src/ol/Map.js';
+import View from '../src/ol/View.js';
+import TileLayer from '../src/ol/layer/Tile.js';
+import IIIF from '../src/ol/source/IIIF.js';
+import IIIFInfo from '../src/ol/format/IIIFInfo.js';
+
+const layer = new TileLayer(),
+    map = new Map({
+      layers: [layer],
+      target: 'map'
+    }),
+    notifyDiv = document.getElementById('iiif-notification'),
+    urlInput = document.getElementById('imageInfoUrl'),
+    displayButton = document.getElementById('display');
+
+function refreshMap(imageInfoUrl) {
+  fetch(imageInfoUrl).then(function(response) {
+    response.json().then(function(imageInfo) {
+      const options = new IIIFInfo(imageInfo).getTileSourceOptions();
+      if (options === undefined || options.version === undefined) {
+        notifyDiv.textContent = 'Data seems to be no valid IIIF image information.';
+        return;
+      }
+      const extent = [0, -options.size[1], options.size[0], 0];
+      options.zDirection = -1;
+      layer.setSource(new IIIF(options));
+      map.setView(new View({
+        resolutions: layer.getSource().getTileGrid().getResolutions(),
+        extent: extent,
```

Comment:
It looks like it would be useful to get the extent from the tile grid instead of having to know how to generate it in this example.  I'd be in support of adding an `@api` annotation to [the `getExtent` method](https://github.com/openlayers/openlayers/blob/e4f377fb4697907fd68a39f7bfcd6497c38c251a/src/ol/tilegrid/TileGrid.js#L237-L240).

If you do that in this branch, you should be able to remove the `extent = [0, -options.size[1], options.size[0], 0]` bit above.

Reply:
Done.